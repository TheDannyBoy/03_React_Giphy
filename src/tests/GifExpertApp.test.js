import { shallow } from "enzyme";
import GifExpertApp from "../components/GifExpertApp";

describe('Testing GifExpertApp.js', () => {
  test('Should create GifExpertApp correctly', () => {
    const wrapper = shallow(<GifExpertApp />)
    expect(wrapper).toMatchSnapshot();
  });

  test('Should show a category list', () => {
    const categories = ['Pok√©mon', 'One Punch'];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />)
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  });
});
