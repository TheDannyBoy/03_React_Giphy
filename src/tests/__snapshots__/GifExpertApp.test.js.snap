// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifExpertApp.js Should create GifExpertApp correctly 1`] = `
<Fragment>
  <h2>
     GifExpertApp 
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Testing GifExpertApp.js Should show a category list 1`] = `
<Fragment>
  <h2>
     GifExpertApp 
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Pokémon"
      key="Pokémon"
    />
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
  </ol>
</Fragment>
`;
